spring-cloud-gcp-data-datastore/src/main/java/org/springframework/cloud/gcp/data/datastore/core/convert/TwoStepsConversions.java

```diff
@@ -49,11 +74,35 @@ public TwoStepsConversions(CustomConversions customConversions) {
 		this.internalConversionService = new DefaultConversionService();
 		this.customConversions = customConversions;
 		this.customConversions.registerConvertersIn(this.conversionService);
+
+		this.internalConversionService.addConverter(BYTE_ARRAY_TO_BLOB_CONVERTER);
+		this.internalConversionService.addConverter(BLOB_TO_BYTE_ARRAY_CONVERTER);
+
 	}
 
 	@Override
 	@SuppressWarnings("unchecked")
-	public <T> T convertOnRead(Object val, Class<?> targetType) {
+	public <T> T convertOnRead(Value val, Class<?> targetType, Class<?> componentType) {
+		Object unwrappedVal = val.get();
+		if (unwrappedVal instanceof List) {
+			try {
+				List<?> elements = ((List<Value>) unwrappedVal)
+						.stream()
+						.map(v -> convertOnReadSingle(v.get(), componentType))
```

Comment:
Would `componentType` be null or unavailable if `targetType` wasn't a list/collection? It seems like if the user had made a mistake and defined his POJO with a singular property but the corresponding datastore column was a List, then we would be on this line of code without a `componentType`. What would the error message be in that case?